src/lib.rs

```diff
@@ -16,3 +16,24 @@ pub mod protocol;
 pub mod setup;
 pub mod tracker;
 pub mod udp;
+
+#[macro_use]
+extern crate lazy_static;
+
+pub mod static_keys {
```

Comment:
In rust, `static` refers to the fact that the memory location is fixed, not that the data at that memory location is immutable.

Originally I chose to name the module `static_foo` as I wanted to namespace the difference between the normal crate root scope and the static variables.

The `lazy_static!` macro only works in the crate root `lib.rs` or `main.rs` files. I wanted to keep the code here as minimal as possible, hence the reason for the `cypto` mod.

Hmm... `generic_keys` or `global_generic_keys` doesn't seem to be as clear to me; maybe: `ephemeral_instance_keys`, to show that these keys are not saved, and are generated on startup?

Reply:
I think the keys are both:

- Generic: because they are not intended to be used only for a single purpose.
- Ephemeral: because there are re-generated on the startup.

Is not the adjective "instance" supposed to describe that the life cycle is attached to the process (instance)?

I think the conflict is I want to explain what you should use them for, and you want to explain how long they last. They both are valid properties, but I think `ephemeral` is more precise than `generic`.

In summary, `ephemeral_instance_keys` is a good name for me.

Reply:
Done, Renamed to `ephemeral_instance_keys`.